utils/update.php

```diff
@@ -68,11 +82,13 @@
         echo "Does the URL point to a directory containing OSM update data?\n\n";
         fail('replication URL not reachable.');
     }
-    $sSetup = CONST_InstallPath.'/utils/setup.php';
-    $iRet = -1;
-    passthru($sSetup.' --create-functions --enable-diff-updates', $iRet);
-    if ($iRet != 0) {
-        fail('Error running setup script');
+    if ($aResult['update-functions']) {
```

Comment:
I'd rather not have more options here.

Reply:
I wanted to account for the case where `setup.php` was run with `--enable-diff-updates`, and therefore it is unnecessary for `update.php` to run it again.

Reply:
I'd be okay with an option `--no-update-functions` if there is really an important use case but the default should be that `setup.php` is executed. I wouldn't consider speed an important use case, though. Rather something along the lines of "I have to put my external server into a special mode to be able to run the --create-functions stuff and don't want to have to do that again".

Reply:
In our case, our web service platform runs the `php` interpreter from a non-standard location (see other comment around this change), so running `…/setup.php` as the command would fail. We therefore always run the initial `setup.php` with the `--enable-diff-updates` argument (with the understanding that the `placex_insert` triggers will be slower). `--no-update-functions` will solve that, though, so I'll go ahead and make that change.